test/injected-script-test.js

```diff
@@ -1,1033 +1,1359 @@
+				}).undefineds,
+				["thisIsUndefined", "obj.thisIsUndefined"],
+				"obj expanded"
+			);
+		});
+	});
+
+	it("updateViewModel", () => {
+		class C extends StacheElement {
+			static get view() {
+				return "<p>{{ this.name }}</p>";
+			}
+
+			static get props() {
+				return {
+					first: { type: type.maybeConvert(String), default: "Kevin" },
+					last: { type: type.maybeConvert(String), default: "McCallister" },
+					get name() {
+						return this.first + " " + this.last;
+					},
+					hobbies: {
+						type: type.Any,
+
+						get default() {
+							return ["running", "jumping"];
+						}
+					}
+				};
+			}
+		}
+
+		customElements.define("a-pp-2", C);
+
+		const el = new C().render();
+
+		assert.deepEqual(
+			devtools.getSerializedViewModelData(el, {
+				expandedKeys: ["hobbies"]
+			}).viewModelData,
+			{
+				first: "Kevin",
+				last: "McCallister",
+				name: "Kevin McCallister",
+				hobbies: { 0: "running", 1: "jumping" }
+			},
+			"default viewmodel data"
+		);
+
+		devtools.updateViewModel(el, [
+			{ type: "set", key: "first", value: "Marty" },
+			{ type: "set", key: "last", value: "McFly" }
+		]);
+
+		assert.deepEqual(
+			devtools.getSerializedViewModelData(el, {
+				expandedKeys: ["hobbies"]
+			}).viewModelData,
+			{
+				first: "Marty",
+				last: "McFly",
+				name: "Marty McFly",
+				hobbies: { 0: "running", 1: "jumping" }
+			},
+			"set works"
+		);
+
+		devtools.updateViewModel(el, [{ type: "delete", key: "last" }]);
+
+		assert.deepEqual(
+			devtools.getSerializedViewModelData(el, {
+				expandedKeys: ["hobbies"]
+			}).viewModelData,
+			{
+				first: "Marty",
+				last: undefined,
+				name: "Marty undefined",
+				hobbies: { 0: "running", 1: "jumping" }
+			},
+			"delete works"
+		);
+
+		devtools.updateViewModel(el, [
+			{
+				type: "splice",
+				key: "hobbies",
+				index: 0,
+				deleteCount: 1,
+				insert: ["skipping"]
+			}
+		]);
+
+		assert.deepEqual(
+			devtools.getSerializedViewModelData(el, {
+				expandedKeys: ["hobbies"]
+			}).viewModelData,
+			{
+				first: "Marty",
+				last: undefined,
+				name: "Marty undefined",
+				hobbies: { 0: "skipping", 1: "jumping" }
+			},
+			"splice works"
+		);
+	});
+
+	it("getViewModelKeys", () => {
+		class C extends StacheElement {
+			static get view() {
+				return "<p>{{ this.name }}</p>";
+			}
+
+			static get props() {
+				return {
+					first: { type: String, default: "Kevin" },
+					last: { type: String, default: "McCallister" },
+					get name() {
+						return this.first + " " + this.last;
+					}
+				};
+			}
+		}
+
+		customElements.define("a-pp-3", C);
+
+		const el = new C().initialize();
+
+		assert.deepEqual(
+			devtools.getViewModelKeys(el),
+			["first", "last", "name"],
+			"gets viewmodel keys"
+		);
+
+		const list = new ObservableArray([{ one: "two" }, { three: "four" }]);
+		assert.deepEqual(
+			devtools.getViewModelKeys(list),
+			["0", "1"],
+			"ignore _ keys for ObservableArray"
```

Comment:
Just curious... are there `_` keys for ObservableArray? I think we're using Symbols for everything. It might be more useful to keep this test using DefineList.

Reply:
I didn't see any `_` keys. Changed it to use `DefineList` as suggested.